### PR TITLE
Add css selectors

### PR DESCRIPTION
**Pull request by MariiaNebesnova**
- added xpath-to-css library
- added "output" property to Locator, that is calculated by root selectors, and reworked usage of getLocator() utility function
- added interface for selecting locator type for PO and Locator
- unit tests added

### CODE REVIEW

File: src/__tests__/pageObject/__mocks__/selectLocatorsByPageObject.mock.ts [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+          children: [],
+          generate: true,
+        },
+      },
+      ids: ["7524916072510597399809892823_0", "2075611903510597386448924232_0", "4138940493550098806301857686_0"],
+    },
+  },
+});
+
+export const result = (pageObject: PageObject) => [
```
Comment (arslanbekova) [inline src/__tests__/pageObject/__mocks__/selectLocatorsByPageObject.mock.ts:132]: let's add a verb to the function name

File: src/__tests__/pageObject/__mocks__/selectLocatorsByPageObject.mock.ts [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+  library: "MUI",
+  pathname: "/jdi-light/index.html",
+  search: "",
+  origin: "https://jdi-testing.github.io",
+  pageData: "",
+  locatorType: LocatorType.cssSelector,
+  locators: ["7524916072510597399809892823_0", "2075611903510597386448924232_0", "4138940493550098806301857686_0"],
+};
+
+export const state = (_pageObject: PageObject) => ({
```
Comment (arslanbekova) [inline src/__tests__/pageObject/__mocks__/selectLocatorsByPageObject.mock.ts:42]: let's add a verb to the function name

File: src/features/locators/utils/locatorOutput.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import xPathToCss from "xpath-to-css";
+import { LocatorType } from "../../../common/types/locatorType";
+import { LocatorValue } from "../types/locator.types";
+
+export const getLocatorByType = (xpath: string, locatorType?: LocatorType) => {
+  try {
+    return locatorType === LocatorType.cssSelector ? xPathToCss(xpath) : xpath;
+  } catch (error) {
+    debugger;
```
Comment (arslanbekova) [inline src/features/locators/utils/locatorOutput.ts:9]: do we need it?

File: src/features/locators/utils/locatorOutput.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  }
+};
+
+export const getXPathbyPriority = ({ fullXpath, robulaXpath, customXpath }: LocatorValue) => {
```
Comment (arslanbekova) [inline src/features/locators/utils/locatorOutput.ts:13]: typo - `getXPathByPriority`

File: src/features/locators/utils/locatorsTreeUtils.ts [hunk 5]
```diff
@@ -13,7 +12,7 @@ export const includesSeacrSubstr = (strings: Array<string>, searchString: string
 
 export const applySearch = (element: Locator, seacrhString: string): SearchState => {
   const { locator, type, name } = element;
-  if (includesSeacrSubstr([getLocator(locator), type as string, name], seacrhString)) {
+  if (includesSeacrSubstr([locator.output, type as string, name], seacrhString)) {
```
Comment (arslanbekova) [inline src/features/locators/utils/locatorsTreeUtils.ts:15]: let's fix typo `includesSearchSubstr`

File: src/__tests__/pageObject/__mocks__/selectLocatorsByPageObject.mock.ts (hunk 1 above)
Comment (MariiaNebesnova) [reply src/__tests__/pageObject/__mocks__/selectLocatorsByPageObject.mock.ts]: changed

File: src/features/locators/utils/locatorOutput.ts (hunk 3 above)
Comment (MariiaNebesnova) [reply src/features/locators/utils/locatorOutput.ts]: No, we don't. Thank you! removed

File: src/features/locators/utils/locatorsTreeUtils.ts (hunk 5 above)
Comment (MariiaNebesnova) [reply src/features/locators/utils/locatorsTreeUtils.ts]: fixed

**Review by arslanbekova — APPROVED** [review]
Thank you!

**Pull request merged**